lib/portlayer/exec/commit.go

```diff
@@ -246,6 +216,94 @@ func Commit(op trace.Operation, sess *session.Session, h *Handle, waitTime *int3
 	return nil
 }
 
+// batchBlockOnFunc is a batching routine that batch processes incoming requests.
+// Incoming request signals the batching routine by throwing into a batch channel. When this routine is performing operation on
+// the previous requests, new requests will be batched in the channel, waiting for next iteration to process.
+func batchBlockOnFunc(ctx context.Context, batch chan chan error, operation func(operation trace.Operation) error) {
+	op := trace.FromContext(ctx, "Add container VM to VMGroup dispatch routine")
+
+	for op.Err() == nil {
+		members := make([]chan error, 0, 5) // batching queue
+		var req chan error
+		var ok bool
+
+		// block and wait for first request
+		select {
+		case req, ok = <-batch:
+			if !ok {
+				return
+			}
+			if req == nil {
+				continue
+			}
+			members = append(members, req)
+		case <-op.Done(): // when parent context is cancelled, quit
+			return
+		}
+
+		// fetch batched requests
+	batchFor:
+		for {
+			select {
+			case req, ok := <-batch:
+				if !ok {
+					return // batching channel closed, quit
+				}
+				if req != nil {
+					members = append(members, req)
+				}
+			default:
+				break batchFor // nothing to batch for now, quit waiting
+			}
+		}
+
+		// process requests
+		err := operation(op)
```

Comment:
if it makes testing easier you could pass either the members array or a batch size argument to the `operation`. That way you can assert that total counts align as expected.

Reply:
Can you elaborate on it? 
I'm not sure if it makes sense to increment a `requests_processed` count in the batching function itself here just for testing purposes. If we keep the count in the `operation` lambda passed in, I'm not sure how feasible it is either.
Based on the specific operation we're using (which is cluster reconfigure), `operation` doesn't really care what the requests are or the # of requests..